kinto/views/permissions.py

```diff
@@ -82,6 +82,13 @@ def get_records(self, filters=None, sorting=None, pagination_rules=None,
         # Check settings for every allowed resources.
         from_settings = allowed_from_settings(self.request.registry.settings, principals)
 
+        # If 'bucket_create_principals' is in settings, setup the '/buckets' uri with
+        # permissions and add the permission lookup
+        bucket_creation = from_settings.get('')
+        if bucket_creation:
+            perms_by_object_uri['/buckets'] = bucket_creation
+            perms_descending_tree[''] = {'bucket:create': {'': {'bucket:create'}}}
```

Comment:
Why not do `perms_descending_tree['bucket'] = {'bucket:create': {'bucket': {'bucket:create'}}}` here and remove the if below ?

Reply:
The issue I saw with that is the way you wrote it, `perms_descending_tree['bucket'] = {'bucket:create': {'bucket': {'bucket:create'}}}`, would overwrite the existing `bucket` permissions taken in from `PERMISSIONS_INHERITANCE_TREE`

We could do something like `perms_descending_tree['bucket'][''] = {'bucket:create': {'': {'bucket:create'}}}`, but then `obtained = perms_descending_tree[resource_name][perm]` would get messed up and we'd be back to another if statement.

Reply:
No, you're right, I can do the following and remove the if statement:

`perms_descending_tree['bucket'].update({'bucket:create': {'bucket': {'bucket:create'}}})`

Reply:
Just let me know if that's acceptable and I'll start on the tests.

Reply:
yep!